src/traces/bar/style.js

```diff
@@ -14,7 +14,7 @@ var attributeOutsideTextFont = attributes.outsidetextfont;
 var helpers = require('./helpers');
 
 function style(gd) {
-    var s = d3.select(gd).selectAll('g.barlayer').selectAll('g.trace');
+    var s = d3.select(gd).selectAll('g[class^="barlayer"]').selectAll('g.trace');
```

Comment:
Yes, this is required because we are naming the classes for each layer of `zindex`. See line 224 in `src/plots/cartesian/index.js`: `var className = (_module.layerName || name + 'layer') + (zindex ? '-' + z : '');`


For example, the mock `zindex_basic`:
![Screenshot 2024-03-13 at 16 34 53](https://github.com/plotly/plotly.js/assets/35932204/d29e12bc-4bfe-4aac-87d2-687d3b7d831c)

Reply:
Perhaps we should use space instead of dash before `zindex`?

Reply:
The name depends on the position of the trace in all the range of zindeces. So if we have two scatter traces one with `zindex=-10` and the other `zindex=10` the class names will be `scatterlayer-0` and `scatterlayer-1`.

Reply:
Sounds good. 
Let's rename the used variable name there from `zindex` to `zOrder`.